client/app/services/databricks-data-source.js

```diff
@@ -5,12 +5,24 @@ import { fetchDataFromJob } from "@/services/query-result";
 
 export default {
   ...DataSource,
-  getDatabases: ({ id }) =>
-    axios
-      .get(`api/databricks/databases/${id}`)
-      .then(data => (has(data, "job.id") ? fetchDataFromJob(data.job.id, 300).catch(() => []) : Promise.resolve([]))),
-  getDatabaseTables: (data, databaseName) =>
-    axios
-      .get(`api/databricks/databases/${data.id}/${databaseName}/tables`)
-      .then(data => (has(data, "job.id") ? fetchDataFromJob(data.job.id, 300).catch(() => []) : Promise.resolve([]))),
+  getDatabases: ({ id }, refresh = false) => {
+    const params = {};
+
+    if (refresh) {
+      params.refresh = true;
+    }
+    return axios
+      .get(`api/databricks/databases/${id}`, { params })
+      .then(data => (has(data, "job.id") ? fetchDataFromJob(data.job.id, 300).catch(() => []) : data));
```

Comment:
We still call `fetchDataFromJob` to make it easier to switch to RQ in the future?

Reply:
The idea was on those lines, but I can actually remove it for now :)